### PR TITLE
TMEDIA-645 - Remove unused amp code in oembed

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Remove amp code in oembed 

## Jira Ticket
- [TMEDIA-645](https://arcpublishing.atlassian.net/browse/TMEDIA-645)

## Acceptance Criteria
No reference to `<amp-` functionality in the Oembed component

Non-amp embed html should still work

## Test Steps

1. Checkout this branch `git checkout REFACTOR-remove-unused-amp`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/article-body-block`
3. Go to http://localhost/pf/test-article-page/?_website=the-gazette
4. Scroll down to see embeds like twitter, fb, etc render as before 

Screen Shot 2022-01-27 at 14 18 14
Screen Shot 2022-01-27 at 14 18 05

5. Notice how youtube is wide because of the news-theme-css class. This is due to deliberate logic 
Screen Shot 2022-01-27 at 15 05 55

### Before
- embeds that reference unsupported amp
- 
### After
- no change
- No reference to code that has amp in it

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 1]
```diff
@@ -1,129 +1,25 @@
...
     </div>
   );
 };
 
 Oembed.propTypes = {
   element: PropTypes.object,
-  isAmp: PropTypes.bool,
-  isLeadArt: PropTypes.bool,
-  content: PropTypes.object,
-  metaValue: PropTypes.func,
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:126]: These props weren't used any where in our blocks

File: package.json [hunk 2]
```diff
@@ -81,7 +81,6 @@
     "node-sass": "^4.14.1",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
-    "react-oembed-container": "^1.0.0",
```
Comment (JackHowa) [inline package.json]: remove top-level dependency if no block needs it

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 3]
```diff
@@ -1,129 +1,25 @@
...
-            <amp-vimeo
-              width="1.7"
-              height="1"
-              layout="responsive"
-              data-videoid={element.raw_oembed.video_id}
-            />
-          );
-        default:
-          return (
-            <EmbedContainer markup={element.raw_oembed.html}>
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:92]: remove embedcontainer as it's not being used for what it's intended for of composing embeds along with other markup

https://github.com/humanmade/react-oembed-container

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 4]
```diff
@@ -1,129 +1,25 @@
...
-
 import React from 'react';
-import PropTypes from 'prop-types';
-import EmbedContainer from 'react-oembed-container';
+import PropTypes from '@arc-fusion/prop-types';
 
 const Oembed = ({
   element,
-  // eslint-disable-next-line no-unused-vars
-  content,
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:11]: this prop wasn't used any way

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 5]
```diff
@@ -1,129 +1,25 @@
...
-  isAmp,
-  isLeadArt,
-  // eslint-disable-next-line no-unused-vars
-  metaValue,
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:15]: this prop also wasn't used any way previously

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 6]
```diff
@@ -1,129 +1,25 @@
...
-import PropTypes from 'prop-types';
-import EmbedContainer from 'react-oembed-container';
+import PropTypes from '@arc-fusion/prop-types';
 
 const Oembed = ({
   element,
-  // eslint-disable-next-line no-unused-vars
-  content,
-  isAmp,
-  isLeadArt,
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:13]: these props are never passed in the blocks codebase

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 7]
```diff
@@ -1,129 +1,25 @@
-/* eslint-disable no-case-declarations */
-// TODO: The css classes referenced here need to be defined...
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:2]: remove todo code and associated css classes that aren't defined

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 8]
```diff
@@ -1,129 +1,25 @@
...
-
 import React from 'react';
-import PropTypes from 'prop-types';
-import EmbedContainer from 'react-oembed-container';
+import PropTypes from '@arc-fusion/prop-types';
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx:2]: update arc proptypes so it works with storybook

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx [hunk 9]
```diff
@@ -1,129 +1,25 @@
...
+  // embed-responsive-16by9, etc are custom classes in news-theme-css
+  const responsiveClass = (['youtube', 'vimeo'].indexOf(element.subtype) !== -1) ? 'embed-responsive embed-responsive-16by9' : '';
 
-    return (
-      <EmbedContainer markup={element.raw_oembed.html}>
-        <div dangerouslySetInnerHTML={{ __html: element.raw_oembed.html }} />
-      </EmbedContainer>
-    );
-  };
+  const cssClasses = `${responsiveClass} block-margin-bottom`;
```
Comment (JackHowa) [inline blocks/article-body-block/chains/article-body/_children/oembed.jsx]: still keeping css classnames that are defined in news-theme-css

File: blocks/article-body-block/chains/article-body/_children/oembed.jsx (hunk 3 above)
Comment (JackHowa) [reply blocks/article-body-block/chains/article-body/_children/oembed.jsx:92]: not sure if evaluating this is part of the scope of this ticket, reverting

File: package.json (hunk 2 above)
Comment (JackHowa) [reply package.json]: not sure if evaluating this is part of the scope of this ticket, reverting

File: .husky/pre-push [hunk 10]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
-# this only for targeting canary at the moment 
-# fetching latest changes to ensure the origin/canary is up to date
+# this only for targeting arc-themes-release-version-1.19 at the moment
```
Comment (matthewroach) [inline .husky/pre-push:3]: These changes are in another PR too - should we create a separate PR to address the issue/needs for this?

File: package.json [hunk 11]
```diff
@@ -30,8 +30,8 @@
     "generate:feature": "hygen feature new",
     "generate:chain": "hygen chain new",
     "prepare": "husky install",
-    "test:changed-feature-branch": "jest --changedSince=origin/canary --coverage --passWithNoTests",
-    "lint:changed-feature-branch": "eslint --max-warnings 0 --no-error-on-unmatched-pattern $(git diff --name-only origin/canary | grep -E \"(.js$|.jsx$)\" || echo \"no js/jsx files changed\")",
+    "test:changed-feature-branch": "jest --changedSince=origin/arc-themes-release-version-1.19 --coverage --passWithNoTests",
```
Comment (matthewroach) [inline package.json:33]: These changes are in another PR too - should we create a separate PR to address the issue/needs for this?

**Review by matthewroach — APPROVED** [review]

File: package.json (hunk 11 above)
Comment (JackHowa) [reply package.json:33]: Curious what the takeaways will be for the pr doing this for themes components. This is possibly the wrong place for a long term solution but needed this for getting past linting locally

Comment (matthewroach) [reply package.json:33]: With the move to release based branching - and the speed it takes to lint file - do we need the overhead of trying to only lint the difference? Could we just run the `npm run lint` command on push?

**Pull request merged**